piecrust-uplink/src/abi/state.rs

```diff
@@ -292,6 +296,51 @@ pub fn owner<const N: usize>(contract: ContractId) -> Option<[u8; N]> {
     }
 }
 
+/// Returns given contract's free limit, if the contract exists and if it
+/// has a free limit set.
+pub fn free_limit(contract: ContractId) -> Option<u64> {
+    let contract_id_ptr = contract.as_bytes().as_ptr();
+    unsafe {
+        match ext::free_limit(contract_id_ptr) as usize {
+            0 => None,
+            _ => with_arg_buf(|buf, _| {
+                if buf[0] == 0 {
+                    None
+                } else {
+                    let mut value_bytes = [0; LEN_U64];
+                    value_bytes.copy_from_slice(&buf[1..LEN_U64 + 1]);
+                    Some(u64::from_le_bytes(value_bytes))
+                }
+            }),
+        }
+    }
+}
```

Comment:
Do we need to use the first byte to signal the `None`? It seems we can use only the return of the function.

```suggestion
/// Returns given contract's free limit, if the contract exists and if it
/// has a free limit set.
pub fn free_limit(contract: ContractId) -> Option<u64> {
    let contract_id_ptr = contract.as_bytes().as_ptr();
    unsafe {
        match ext::free_limit(contract_id_ptr) {
            0 => None,
            _ => with_arg_buf(|buf, _| {
                let mut value_bytes = [0; LEN_U64];
                value_bytes.copy_from_slice(&buf[0..LEN_U64]);
                Some(u64::from_le_bytes(value_bytes))
            }),
        }
    }
}
```

Reply:
yes, we convey existence of contract and also existence of the setting, hence, we need to encode an option separately, so the above suggestion is not good IMO